AdvancedTeleport-Bukkit/src/main/java/io/github/niestrat99/advancedteleport/payments/PaymentManager.java

```diff
@@ -105,8 +105,8 @@ public void withdraw(String command, Player player) {
     }
 
     private HashMap<String, Payment> getPayments(String command, Player player) {
-        ConfigurationSection customCosts = NewConfig.get().CUSTOM_COSTS.get();
-        HashMap<String, Payment> payments = new HashMap<>();
+        final var customCosts = NewConfig.get().CUSTOM_COSTS.get();
+        var payments = new HashMap<String, Payment>();
```

Comment:
Still aiming for Java 8 support here for the time being, which does not support var.

Reply:
1.16 is still Java 8, no? I know Paper released their statement on Java 11 around this time but that doesn't include Spigot which is the main platform AT still has to support (for now).

Reply:
I believe 1.16 was fully compatible with java 17, ill have to do some digging to find out

Reply:
Okay so 1.16 does indeed support a maximum java version of 16 meaning we could move the java version to 16 features and maintain it on 17

Reply:
Sorry, I'm a little behind since I was with family and working on an event last night. I got off today and will focus more on my personal projects at least. :) 

I am a little opposed to enforcing a minimum version of Java 16 on servers since I work in server hosting and the place I work at automatically sets Java versions depending on the server version and type being used, i.e. Java 11 for Paper 1.16.5, Java 8 for Spigot and Java 8 for Forge 1.16.5 (including this because of Arclight, which I don't actively encourage but don't discourage either). Because of that, I don't fully know how other popular hosting providers work with this, but looking at plugins that go down this road - looking at you, PlotSquared - are a bit of a pain in the ass to deal with for both us and server owners who just want the plugin to work outside of the box.

We can certainly make Java 16 a minimum version when 1.16.5 eventually wilts out like 1.12.2 and 1.8, but I think we can hold off that for a bit longer.

Reply:
Hmm, that makes sense for the providers which don't allow changing the java version, I hadn't thought about that as I've only ever hosted on local hardware or full dedicated machines, I'll have to go back and double check what i need to change if anything for java 11 but that should at-least allow for some of the newer features.

Reply:
Yeah, I fully understand. It's worth having the extra perspective. I try to aim at making the plugin as easily usable as possible, even if it still needs some cleaning up here and there. Thank you for taking that into account!

Var was introduced in Java 10 iirc.

Reply:
So interesting development is that without setting the java version it 16 at the lowest the paper api and some other dependencies become unresolvable and require a higher java version, While we could compile it to 16 with just 11 features it still wouldn't run on java 11 and would require a minimum of 16

Reply:
I think I did run into that when initially introducing Gradle to the dev environment for AT, I *think* it ran fine on Java 8? Not quite sure. I do love the mess that is MC's Java version management

Reply:
Doesn't seem to work with 8 either, and with more libraries overtime this will become more of an issue

Reply:
Crud :( okay, keep Java 16/17 in place then, I'll keep it maintained at the latest server version support (i.e. 1.18 and 1.19). I can maintain a deprecated branch of AT solely for bug fixes in older versions. Sorry for the trouble.

Reply:
That's fine, there not really much of a way to work around minecrafts absolutely fucked java versions.